rustls/src/client/builder.rs

```diff
@@ -4,21 +4,21 @@ use crate::client::{ClientConfig, ResolvesClientCert};
 use crate::crypto::{CryptoProvider, SupportedKxGroup};
 use crate::key_log::NoKeyLog;
 use crate::suites::SupportedCipherSuite;
-#[cfg(feature = "ring")]
+#[cfg(any(feature = "ring", feature = "aws_lc_rs"))]
```

Comment:
Instead of this, could/should we refactor the module hierarchy so that these things no longer have to be source from the current provider module?

Reply:
I think this would mean losing APIs which I think usefully hide complexity from downstream users, for example:

`.with_client_auth_cert(cert_chain, key_der)` would become something like `.with_client_cert_resolver(Arc::new(rustls::client::AlwaysResolvesClientCert::new(rustls::sign::CertifiedKey::new(cert_chain, rustls::crypto::ring::sign::any_supported_type(priv_key).expect("cannot parse private key")))))`

Reply:
Oh, I see. I'll look into that direction. Thanks!

Reply:
Can this thread be considered resolved by https://github.com/rustls/rustls/pull/1414/commits/68792f97ab20d12c38c465c2307aec144958934d?

Reply:
Can this thread be considered resolved by https://github.com/rustls/rustls/commit/68792f97ab20d12c38c465c2307aec144958934d?